scene.js

```diff
@@ -72,31 +73,41 @@ class Scene {
         mat4.multiply(this.viewMatrix, yRotation, xRotation);
         this.viewMatrix[14] = -translate;
 
-        var firstNode = this.nodes[0];
-
-        drawNodeRecursive(this, firstNode, mat4.create());
+        // draw the scene; note - do not assume a single fixed root
+        for (var ni = 0; ni < this.scene.nodes.length; ni++) {
+            var firstNode = this.nodes[this.scene.nodes[ni]];
+            drawNodeRecursive(this, firstNode, mat4.create());
+        }
 
         // draw to the front buffer
         this.frontBuffer.drawImage(this.backBuffer, 0, 0);
     }
 
-    loadImage(imageInfo, gl) {
+    loadImage(imageInfo, gl,mesh) {
         var scene = this;
         var image = new Image();
         this.pendingTextures++;
         image.src = imageInfo.uri;
         image.onload = function() {
             var texture = gl.createTexture();
             var glIndex = gl.TEXTURE0 + imageInfo.samplerIndex;  // gl.TEXTUREn enums are in numeric order.
-            gl.activeTexture(glIndex);
-            gl.bindTexture(gl.TEXTURE_2D, texture);
-
+            
+            // save this information in the material - we will bind it later when we come to drawing this mesh
+            if (!defined(mesh.material.samplers)) 
+              mesh.material.samplers = new Array();
```

Comment:
Braces and clean indentation are preferred here as well.  But first, consider moving all this code into the mesh class, so that meshes can share images.